Version 0.4

SebastianSemper opened the pull request:
Let's get this thing going.

SebastianSemper committed:
Add HFSS demo

SebastianSemper committed:
change copyright year

SebastianSemper committed:
remove print

SebastianSemper committed:
Fix bug in narrowband transform

SebastianSemper committed:
added importer to use array pattern data of the form ele x azi x elem x pol x freq

SebastianSemper commented on eadf/importers.py:
```diff
@@ -186,3 +189,53 @@ def fromAngleListData(
 
     return EADF(arrInter, arrAzi, arrCoEle, arrFreqData, arrPos)
+
+def loadData(
```
The function name is completely void of information. It should be changed to something more meaningful.

SebastianSemper commented on eadf/importers.py:
```diff
@@ -186,3 +189,53 @@ def fromAngleListData(
+
+def loadData(
+    path: str,
+    numPor = 0,
```
why is it the number of ports, and not an array of port indices? why is this feature only there for the antenna ports and not also for frequency and polarisation? also the specification of the default value is not pep8-conform.

pawarsp replied:
Agree that the user should be able to choose the specific antenna ports, polarization and frequency points.

SebastianSemper commented on eadf/importers.py:
```diff
@@ -186,3 +189,53 @@ def fromAngleListData(
+    """
+    This format defines angular data over uniformly sampled
+    ele, azi, elem, pol, and freq. Using the importer it is possible to choose
+    the number of ports to derive the EADF by supplying the numPor value.
```
why must it be uniformly sampled along frequency and polarisation? what does that even mean? also, there is no description about the internal structure of the expected matfile. how would someone be able to produce a .mat-file that can be processed by this importer? however, this is crucially necessary.

pawarsp replied:
by this did you mean that the details about the measurement method itself are lacking?

SebastianSemper commented on eadf/importers.py:
```diff
@@ -186,3 +189,53 @@ def fromAngleListData(
+    numPorData = (data["pattern"][0][0][0][0][2][1]).size
+
+    if (numPor > numPorData) or (numPor < 1):
+        numPor = 9
```
what? this looks wrong. either throw an error if the function parameters are invalid, or at least issue a warning, that we did something different than the user specified.

pawarsp replied:
that is certainly incorrect.

SebastianSemper reviewed (commented):
the current state of loadData is not acceptable. not only does it fail the pycodestyle checks, but it also is not well documented and does not possess enough functionality as one might wish.

SebastianSemper commented:
No, I mean that as a user one does not know what should be in the mat-file
such that the importer works. What objects have to be there in Matlab? How
should they be named? Their dimensions? Their datatype? In general: what is
the exakt file format description for the files your importer imports?

Sankalp Pawar <notifications@github.com> schrieb am Do., 6. Feb. 2020,
16:52:

> *@StellarSankalp* commented on this pull request.
> ------------------------------
>
> In eadf/importers.py
> <https://github.com/EMS-TU-Ilmenau/EADF/pull/34#discussion_r375918497>:
>
> > +    This format defines angular data over uniformly sampled
> +    ele, azi, elem, pol, and freq. Using the importer it is possible to choose
> +    the number of ports to derive the EADF by supplying the numPor value.
>
> by this did you mean that the details about the measurement method itself
> are lacking?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/EMS-TU-Ilmenau/EADF/pull/34?email_source=notifications&email_token=AAINSRUTOU3NAGDECGJWCA3RBQW5BA5CNFSM4KL6LYZ2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCURC2EA#discussion_r375918497>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAINSRRJRNFDHEZ3ZAXR7O3RBQW5BANCNFSM4KL6LYZQ>
> .
>

SebastianSemper committed:
Add Makefile

SebastianSemper committed:
Add back Fourier Import, add demo

SebastianSemper committed:
Fix bug in fourier importer, add test

SebastianSemper committed:
Fix Bug in AngleListTest, add testing to makefile

MichaelDoebereiner committed:
added importer for TUI MatLab arraydef structure

MichaelDoebereiner committed:
Merge branch 'dev0.4' of https://github.com/EMS-TU-Ilmenau/EADF into dev0.4

SebastianSemper committed:
fix fourier demo, blackify

SebastianSemper committed:
add black badge

SebastianSemper committed:
fix black badge

Commit:
fix bug in periodification, add tornado to requs

pawarsp committed:
extend the matfile importer

SebastianSemper committed:
added importer for mat file data

SebastianSemper committed:
added mat file importer

MichaelDoebereiner committed:
bug fix for arraydef importer
design changes

MichaelDoebereiner committed:
added plot of pattern and Fourier coefficients for one element

pawarsp committed:
added mat file importer

MichaelDoebereiner committed:
bug fixes for Matlab arraydef importer

SebastianSemper committed:
Add some optimizations

1. we have a new property: _einsumArgs, which is a dictionary
   of keyworded arguments we pass to einsum. this changes depending
   on whether we use cupy or numpy
2. some nested for loops were removed in the preprocessing
   routines.

SebastianSemper committed:
Merge remote-tracking branch 'origin/matfile' into dev0.4

SebastianSemper committed:
blackify

SebastianSemper committed:
Merge branch 'moarspeed' into dev0.4

pawarsp committed:
added kwargs to importer for import of pattern as .mat file, also added demo for it

pawarsp committed:
improved styling issues

pawarsp committed:
used make black

SebastianSemper committed:
Merge branch 'matfileImporter' into dev0.4

SebastianSemper committed:
describe structure of gradients and hessians

SebastianSemper committed:
fix wrong alignment of pattern

SebastianSemper committed:
try another fix

SebastianSemper committed:
finally fix it?

SebastianSemper committed:
pfew...

SebastianSemper committed:
replace einsum with faster methods

* speed up narroband

* introduce blocking in the core function

* determine the optimal block size

* add stylechecking to makefile

MichaelDoebereiner committed:
fix of flipfix

SebastianSemper committed:
Remove Fourier Data Importer

SebastianSemper committed:
Merge branch 'flipfix' into dev0.4

SebastianSemper committed:
Refactor backend selection, add lowmem property

1. The backend is now selected via EADF_BACKEND=cupy/numpy
2. One can set the array.lowMem flag, which generates the
   necessary blocks of the transformation matrices only on
   demand. this is especially useful, if one has a gpu with
   little memory

pawarsp committed:
added importer for narrowband pattern along with related demo

SebastianSemper committed:
update docu

SebastianSemper committed:
add asNumpy to backend, delete unnecessary demos

pawarsp committed:
revised importers for importing pattern data

SebastianSemper committed:
Add docu for the backends

pawarsp committed:
updated importers.py and added demo patternNBandImport.py

pawarsp committed:
updated file and variable names for pattern file importers and respective demos

SebastianSemper committed:
Extend Importers

1. Added input checks for all file-based importers
2. changed the interface of some importers to return
   a normal eadf object and not tuple, adjusted
   the demos accordingly.

SebastianSemper committed:
Merge remote-tracking branch 'origin/matFileImporterNarrowBand' into dev0.4

SebastianSemper committed:
Merge branch 'lowmem' into dev0.4

pawarsp committed:
updated core.py and eadf.py to enable block processing, and updated demos to include lowmem feature

pawarsp committed:
improved Travis error

SebastianSemper committed:
remove unneeded properties, fix function calls

pawarsp committed:
updated core.py and eadf.py to enable block processing, and updated d… (#47)

* updated core.py and eadf.py to enable block processing, and updated demos to include lowmem feature

* improved Travis error

* remove unneeded properties, fix function calls

SebastianSemper committed:
add image to inputviz dmeo

SebastianSemper committed:
add, modify, delete demos

SebastianSemper committed:
Merge branch 'dev0.4' into doc0.4

SebastianSemper committed:
add output of narrowband demo

SebastianSemper committed:
iterate over the backend

SebastianSemper committed:
add core tests

SebastianSemper committed:
Extend Tests, Fix Bugs, Move Plots

SebastianSemper committed:
Fix travis.yml and setup.py

SebastianSemper committed:
fix travis.yml

SebastianSemper committed:
Add lowmem env variable

SebastianSemper committed:
fix bug in transform calls

SebastianSemper committed:
add dtype env variable

SebastianSemper committed:
extend build jobs

SebastianSemper committed:
allow nightly to fail, fast finish

SebastianSemper committed:
remove 3.4 and 3.5, add badges

SebastianSemper committed:
remove OSX

ChristophWWagner committed:
Correct typos in docu

ChristophWWagner committed:
Fix encoding problem with german 'Umlaut's

ChristophWWagner committed:
Add the notorious makefile

... featuring useful commands such as
  `make install/uninstall/styleCheck/codeStats/...`

Mitigate line PEP8 continuation errors

Corrected makefile styleCheck configuration

The Travis CI run contained some information about the style guidelines
for this project, which it communicated by failing the extension :)
Therefore the makefile styleCheck configuration was adapted to match
that of the Travis CI suite and errors were corrected.

ChristophWWagner committed:
Add HFSS importer

Unfortunately, verification of the imported data was not completed
fully yet, due to another bug (see #52), which prevents EADF object
initialization for the data contained in the available HFSS exports.
Stay tuned...

SebastianSemper committed:
fix bug in plot routines

SebastianSemper committed:
move input checks to different file

ChristophWWagner committed:
Fix codacy issues

ChristophWWagner committed:
Travis CI: Limit deployment to master branch only

ChristophWWagner committed:
Reintroduce exception raises for errors (WIP)

This commit only scopes the `EADF.eadf` submodule. It is suggested to check
out the modifications of this unit, and the corresponding tests in
`test_eadf.py`, prior to extending the modifications to the whole package.

The modifications in this commit tackle issue #56 and propose to apply the
approach detailed here throughout the whole package. Please discuss it in
the thread to issue #56.

Modifications:
--------------
 - Instances of `logging.error` were replaced by raising exceptions
 - A new `TestCase` baseclass was introduced in `test_eadf.py` which offers
   a new method, `assertRaises With Message` that is capable of running some
   code as a contextmanager and check if an exception of proper king was
   raised therein, producing an intended error message. The test fails if
   either no exception was thrown, an exception of the wrong type was thrown,
   or the error message does not match the requirement.
   Checking the exception message is optional.
 - All instances of mocking `logging.error` within `test_eadf.py` was replaced
   by the new assertion mechanisms for exceptions.

ChristophWWagner committed:
Extend EADF.__checkInputData()

This commit tackles issues #53 and #55.

 - Remove code duplicates for checks of input parameter shape and gridding.
 - Introduce handling of SI unit types by extending input parameters to pint
   Quantities.
 - Refactoring the structure of `EADF.__checkInputData(...)` to enable
   modifying the input arrays prior to object initialization. For this,
   passing additional keyworded arguments `EADF.__init__(..., **options)` is
   possible by handing `**options` through to `EADF.__checkInputData()`.

ChristophWWagner committed:
Modify keyworded argument declaration in docs

ChristophWWagner committed:
Correct HFSS function name for polar phase type

ChristophWWagner committed:
Fix problem of dual unit registry

ChristophWWagner committed:
Add preprocessing capabilities to EADF constructor

 - Allows to modify the support values for the angular and frequency
   axes by passing a function to the `**options`:
     `modify_support_CoEle` for the Co-Elevation axis
     `modify_Azi` for the Azimuth axis
     `modify_support_Freq` for the Frequency axis.
   Only the support vector is altered, not the tensor data itself.
   For example, this is useful for adjusting the specification of
   angles in Elevation to Co-Elevation angles.
 - Truncate along angular or frequency axes by providing a (min,max)
   tuple, indicating the cropping range, to the `**options`:
     `truncate_CoEle` for the Co-Elevation axis
     `truncate_Azi` for the Azimuth axis
     `truncate_Freq` for the Frequency axis
 - Sort the tensor data along the angular and frequency axes if
   `sort=True` is given in `**options`. Also sorts the support

ChristophWWagner committed:
Fix missing patch definition in test_eadf.py

ChristophWWagner committed:
Provide HFSS importer usage example

ChristophWWagner committed:
Refactor exceptions in importers and test_importers

ChristophWWagner committed:
Add stylecheck to make test

ChristophWWagner committed:
Refactor exception handling in arrays and test_arrays

ChristophWWagner committed:
Refactor exception raising. Solves #56

SebastianSemper committed:
Fix minor bug in eval

SebastianSemper committed:
Merge branch 'doc0.4' into dev0.4

SebastianSemper committed:
blackify, fix merge issues

sebarn committed:
add missing pi in derivatives

ChristophWWagner committed:
Fix quantity-related issues

ChristophWWagner committed:
Increase verbsity of HFSS importer

When importing fails, HFSS importer now reports which parameters and
types were actually present in the data (all found), assisting in
identifying problems when matching data sets.

ChristophWWagner committed:
Resolve inconsistency in renaming <EADF>.unitData

ChristophWWagner committed:
Fix inconsistency of tensor and EADF.__init__() layout

Swapped `CoEle`and `Azi` argument positions. Resolves #58.
Also fixes some of the code quality issues in this branch.

sebarn committed:
extend docu, fix hessian, blackify
